packages/python-packages/api-stub-generator/apistub/nodes/_function_node.py

```diff
@@ -0,0 +1,218 @@
+import logging
+import inspect
+import astroid
+import operator
+from inspect import Parameter
+from ._docstring_parser import DocstringParser, TypeHintParser
+from ._base_node import NodeEntityBase, get_qualified_name
+from ._argtype import ArgType
+
+
+KW_ARG_NAME = "**kwargs"
+
+
+class FunctionNode(NodeEntityBase):
+    """Function node class represents parsed function signature.
+    Keyword args will be parsed and added to signature if docstring is available.
+    :param str: namespace
+    :param NodeEntityBase: parent_node
+    :param function: obj
+    :param bool: is_module_level
+    """
+
+    def __init__(self, namespace, parent_node, obj, is_module_level=False):
+        super().__init__(namespace, parent_node, obj)
+        self.annotations = []
+        self.args = []
+        self.return_type = None
+        self.namespace_id = self.generate_id()
+        # Set name space level ID as full name
+        # Name space ID will be later updated for async methods
+        self.full_name = self.namespace_id
+        self.is_class_method = False
+        self.is_module_level = is_module_level
+        # Some of the methods wont be listed in API review
+        # For e.g. ABC methods if class implements all ABC methods
+        self.hidden = False
+        self._inspect()
+
+    def _inspect(self):
+        logging.debug("Processing function {0}".format(self.name))
+        code = inspect.getsource(self.obj).strip()
+        # We cannot do "startswith" check here due to annotations or decorators present for functions
+        self.is_async = code.__contains__("async def")
```

Comment:
wondering if you could do something like:
```
self.is_async = "async def" in code
```
There's usually a way to avoid calling dunder methods directly.